### PR TITLE
Support verify_fips_enable test case for CBLMariner

### PR DESCRIPTION
**Pull request by suresh-thelkar**

### CODE REVIEW

File: microsoft/testsuites/security/fips.py [hunk 1]
```diff
@@ -26,20 +27,57 @@ class Fips(TestSuite):
...
     )
     def verify_fips_enable(self, log: Logger, node: Node) -> None:
-        result = node.execute("command -v fips-mode-setup", shell=True)
-        if result.exit_code != 0:
-            raise SkippedException(
-                "Command not found: fips-mode-setup. "
-                f"Please ensure {node.os.name} supports fips mode."
-            )
+        if isinstance(node.os, CBLMariner):
+            result = node.execute("sudo cat /proc/sys/crypto/fips_enabled")
```
Comment (LiliDeng) [inline microsoft/testsuites/security/fips.py]: use tool Cat.  node.tools[Cat]

Comment (suresh-thelkar) [reply microsoft/testsuites/security/fips.py]: Thanks @LiliDeng  for the review. I have implemented your review comments and also fixed few more warnings. Please take a look.

File: microsoft/testsuites/security/fips.py [hunk 2]
```diff
@@ -26,20 +30,64 @@ class Fips(TestSuite):
...
     def verify_fips_enable(self, log: Logger, node: Node) -> None:
-        result = node.execute("command -v fips-mode-setup", shell=True)
-        if result.exit_code != 0:
-            raise SkippedException(
-                "Command not found: fips-mode-setup. "
-                f"Please ensure {node.os.name} supports fips mode."
-            )
+        if isinstance(node.os, CBLMariner):
+            # check If It is a non-FIPS image
+            if node.execute("rpm -qa | grep dracut-fips").exit_code != 0:
```
Comment (LiliDeng) [inline microsoft/testsuites/security/fips.py]: I think this command needs sudo

File: microsoft/testsuites/security/fips.py [hunk 3]
```diff
@@ -26,20 +30,64 @@ class Fips(TestSuite):
...
+
+                if "1" != result.stdout:
+                    raise LisaException(
+                        "fips is not enabled properly. "
+                        f"Please ensure {node.os.name} has fips turned "
+                        "on by default."
+                    )
+
+                result = node.execute("openssl md5")
+                print(result)
```
Comment (LiliDeng) [inline microsoft/testsuites/security/fips.py]: remove debug line

File: microsoft/testsuites/security/fips.py [hunk 4]
```diff
@@ -26,20 +30,60 @@ class Fips(TestSuite):
...
+                # md5 should not work If It is a FIPS image
+                # Following the output of the above command
+                # Error setting digest
+                # 131590634539840:error:060800C8:digital envelope routines:EVP_DigestInit_ex:disabled for FIPS:crypto/evp/digest.c:135:
+                if result.exit_code != 0:
+                    pattern = re.compile(r"Error setting digest\n(.+?disabled for FIPS:crypto.+?)\n")
+                    if not get_matched_str(result.stdout, pattern):
+                        raise LisaException(
+                            "openssl is not operating under fips mode."
```
Comment (LiliDeng) [inline microsoft/testsuites/security/fips.py]: `is not operating` is expected or not? please give the clear exception message

File: microsoft/testsuites/security/fips.py [hunk 5]
```diff
@@ -26,20 +30,60 @@ class Fips(TestSuite):
...
+                        )
+                else:
+                    raise LisaException(
+                        "Not a valid FIPS image."
```
Comment (LiliDeng) [inline microsoft/testsuites/security/fips.py]: give a clear message

File: microsoft/testsuites/security/fips.py [hunk 6]
```diff
@@ -26,20 +30,60 @@ class Fips(TestSuite):
...
+                    )
+
+                result = node.execute("openssl md5")
+                print(result)
+                # md5 should not work If It is a FIPS image
+                # Following the output of the above command
+                # Error setting digest
+                # 131590634539840:error:060800C8:digital envelope routines:EVP_DigestInit_ex:disabled for FIPS:crypto/evp/digest.c:135:
+                if result.exit_code != 0:
+                    pattern = re.compile(r"Error setting digest\n(.+?disabled for FIPS:crypto.+?)\n")
```
Comment (LiliDeng) [inline microsoft/testsuites/security/fips.py]: put pattern in class level

File: microsoft/testsuites/security/fips.py (hunk 2 above)
Comment (suresh-thelkar) [reply microsoft/testsuites/security/fips.py]: No need of sudo here as we are just querying for the package existence only. I have verified it.

File: microsoft/testsuites/security/fips.py (hunk 4 above)
Comment (suresh-thelkar) [reply microsoft/testsuites/security/fips.py]: I have added more detailed messages. Please take a look at the latest commit.

File: microsoft/testsuites/security/fips.py (hunk 5 above)
Comment (suresh-thelkar) [reply microsoft/testsuites/security/fips.py]: Added a descriptive message.

File: microsoft/testsuites/security/fips.py (hunk 6 above)
Comment (suresh-thelkar) [reply microsoft/testsuites/security/fips.py]: Implemented this as per your suggestion. Please take a look at the latest commit.

**Comment by suresh-thelkar** [comment]
@LiliDeng, I have addressed your comments. Please take a look at the latest commit.

**Comment by LiliDeng** [comment]
looks good, will merge after next release

**Comment by suresh-thelkar** [comment]
> looks good, will merge after next release

Thank you very much for all the reviews and approval.

**Pull request merged**